DokanNet/DokanHelper.cs

```diff
@@ -0,0 +1,160 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace DokanNet
+{
+    public class DokanHelper
+    {
+        /// <summary>
+        /// Matches zero or more characters until encountering and matching the final . in the name.
+        /// </summary>
+        const Char DOS_STAR = '<';
+
+        /// <summary>
+        /// Matches any single character or, upon encountering a period or end
+        /// of name string, advances the expression to the end of the set of
+        /// contiguous DOS_QMs.
+        /// </summary>
+        const Char DOS_QM = '>';
+
+        /// <summary>
+        /// Matches either a period or zero characters beyond the name string.
+        /// </summary>
+        const Char DOS_DOT = '"';
+
+        /// <summary>
+        /// Matches zero or more characters.
+        /// </summary>
+        const Char ASTERISK = '*';
+
+        /// <summary>
+        /// Matches a single character.
+        /// </summary>
+        const Char QUESTION_MARK = '?';
+
+        private static Char[] CharsThatMatchEmptyStringsAtEnd = new Char[] { DOS_DOT, DOS_STAR, ASTERISK };
+
+        /// <summary>
+        /// Check whether <paramref name="Name">Name</paramref> matches <paramref name="Expression">Expression</paramref>.
+        /// </summary>
+        /// <remarks>
+        /// This method is mainly used in <see cref="IDokanOperations.FindFilesWithPattern"/> to filter a list of possible files.
+        /// <seealso cref="http://msdn.microsoft.com/en-us/library/ff546850(v=VS.85).aspx"/> 
+        /// </remarks>
+        /// <param name="Expression">The matching pattern. Can contain: ?, *, &lt;, &quot;, &gt;.</param>
+        /// <param name="Name">The string that will be tested.</param>
+        /// <param name="IgnoreCase">When set to true a case insensitive match will be performed.</param>
+        /// <returns>Returns true if Expression match Name, false otherwise.</returns>
+        /// <example>"F0_&lt;&quot;*" match "f0_001.txt"</example>

```

Comment:
Doxygen do ignore the `<example>` tag, see https://www.stack.nl/~dimitri/doxygen/manual/xmlcmds.html. Move the example to some other tag.